Userland/Applications/PDFViewer/PDFViewerWidget.cpp

```diff
@@ -298,6 +302,41 @@ ErrorOr<void> PDFViewerWidget::initialize_menubar(GUI::Window& window)
     return {};
 }
 
+NonnullRefPtr<Gfx::Bitmap const> PDFViewerWidget::render_thumbnail_for_rendered_page(u32 page_index)
+{
+    int rect_square_size = 96;
+    Gfx::IntSize thumbnail_size(0, 0);
+    auto rendered_page = m_viewer->get_rendered_page(page_index).value();
```

Comment:
I suppose with lazy thumbnails it doesn't matter all that much, but I think this renders the PDF to a full-size bitmap and then scales it down – rendering at a smaller scale in the first place might be faster.